src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java

```diff
@@ -0,0 +1,234 @@
+            return data;
+        } else if (!APPROVED.getCode().equals(approvalStatus)) {
+            throw new ValidationException(UNKNOWN_NOC_APPROVAL_STATUS);
+        }
+
+        applyDecision(caseDetails, changeOrganisationRequestField, caseRoleId);
+
+        nullifyNode(changeOrganisationRequestField);
+        return data;
+    }
+
+    private void applyDecision(CaseDetails caseDetails, JsonNode changeOrganisationRequestField, String caseRoleId) {
+        JsonNode orgPolicyNode = caseDetails.findOrganisationPolicyNodeForCaseRole(caseRoleId);
+
+        JsonNode organisationToAddNode = changeOrganisationRequestField.get(ORGANISATION_TO_ADD);
+        JsonNode organisationToRemoveNode = changeOrganisationRequestField.get(ORGANISATION_TO_REMOVE);
+        Organisation organisationToAdd = objectMapper.convertValue(organisationToAddNode, Organisation.class);
+        Organisation organisationToRemove = objectMapper.convertValue(organisationToRemoveNode, Organisation.class);
+
+        if (organisationToAdd == null || isNullOrEmpty(organisationToAdd.getOrganisationID())) {
+            applyRemoveRepresentationDecision(caseDetails.getReference(), orgPolicyNode, organisationToRemove);
+        } else {
+            applyAddOrReplaceRepresentationDecision(caseDetails.getReference(), caseRoleId, orgPolicyNode,
+                    organisationToAddNode, organisationToAdd, organisationToRemove);
+        }
+    }
+
+    private void validateCorFieldOrganisations(JsonNode changeOrganisationRequestField) {
+        if (!changeOrganisationRequestField.has(ORGANISATION_TO_ADD)
+                || !changeOrganisationRequestField.has(ORGANISATION_TO_REMOVE)) {
+            throw new ValidationException(COR_MISSING_ORGANISATIONS);
+        }
+    }
+
+    private void applyAddOrReplaceRepresentationDecision(String caseReference,
+                                                         String caseRoleId,
+                                                         JsonNode orgPolicyNode,
+                                                         JsonNode organisationToAddNode,
+                                                         Organisation organisationToAdd,
+                                                         Organisation organisationToRemove) {
+        setOrgPolicyOrganisation(orgPolicyNode, organisationToAddNode);
+        assignAccessToOrganisationUsers(caseReference, organisationToAdd, caseRoleId);
+
+        if (organisationToRemove != null && !isNullOrEmpty(organisationToRemove.getOrganisationID())) {
+            removeOrganisationUsersAccess(caseReference, organisationToRemove, caseRoleId);
+        }
+    }
+
+    private void applyRemoveRepresentationDecision(String caseReference,
+                                                   JsonNode orgPolicyNode,
+                                                   Organisation organisationToRemove) {
+        nullifyNode(orgPolicyNode.get(ORGANISATION));
```

Comment:
instead of nullifyNode, can't we remove this node?

Reply:
The design states we should nullify all the fields (otherwise agree with you!)